### ISSUE TITLE
storage/concurrency: test failure TestLockTableConcurrentRequests/concurrency_16

### ISSUE THREAD

**Issue by nvanbenschoten** [cockroachdb/cockroach#45151]
See https://teamcity.cockroachdb.com/viewLog.html?buildId=1753074&buildTypeId=Cockroach_UnitTests&fromExperimentalUI=true.

Under testrace:
```
=== RUN   TestLockTableConcurrentRequests/concurrency_16
fatal error: sync: unlock of unlocked mutex

goroutine 125088 [running]:
runtime.throw(0x270d4af, 0x1e)
	/usr/local/go/src/runtime/panic.go:774 +0x72 fp=0xc001b71808 sp=0xc001b717d8 pc=0xbe0752
sync.throw(0x270d4af, 0x1e)
	/usr/local/go/src/runtime/panic.go:760 +0x35 fp=0xc001b71828 sp=0xc001b71808 pc=0xbe06d5
sync.(*Mutex).unlockSlow(0xc0020bcae0, 0x14e4ffffffff)
	/usr/local/go/src/sync/mutex.go:196 +0x10d fp=0xc001b71850 sp=0xc001b71828 pc=0xc23ecd
sync.(*Mutex).Unlock(0xc0020bcae0)
	/usr/local/go/src/sync/mutex.go:190 +0x7b fp=0xc001b71878 sp=0xc001b71850 pc=0xc23dab
github.com/cockroachdb/cockroach/pkg/util/syncutil.(*Mutex).Unlock(0xc0020bcae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/mutex_sync_race.go:36 +0x55 fp=0xc001b71898 sp=0xc001b71878 pc=0x11366f5
runtime.call32(0x0, 0x2787760, 0xc001b71ba8, 0x800000008)
	/usr/local/go/src/runtime/asm_amd64.s:539 +0x3b fp=0xc001b718c8 sp=0xc001b71898 pc=0xc0ffdb
panic(0x2657380, 0xc0001000c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2 fp=0xc001b71958 sp=0xc001b718c8 pc=0xbe0292
runtime.goPanicIndex(0x2, 0x2)
	/usr/local/go/src/runtime/panic.go:75 +0xa3 fp=0xc001b719a0 sp=0xc001b71958 pc=0xbde923
github.com/cockroachdb/cockroach/pkg/storage/spanset.(*SpanSet).GetSpans(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/spanset/spanset.go:146
github.com/cockroachdb/cockroach/pkg/storage/concurrency.(*lockTableImpl).findNextLockAfter(0xc00034e000, 0xc0020bca80, 0xc001b71e00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/concurrency/lock_table.go:1617 +0xb13 fp=0xc001b71b60 sp=0xc001b719a0 pc=0x1bbb1c3
github.com/cockroachdb/cockroach/pkg/storage/concurrency.(*lockTableGuardImpl).CurState(0xc0020bca80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/concurrency/lock_table.go:295 +0x132 fp=0xc001b71be0 sp=0xc001b71b60 pc=0x1baaa42
github.com/cockroachdb/cockroach/pkg/storage/concurrency.doWork(0x2997c00, 0xc0002af180, 0xc00103d380, 0xc000153860, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/concurrency/lock_table_test.go:479 +0x1f0 fp=0xc001b71f08 sp=0xc001b71be0 pc=0x1bbdc60
github.com/cockroachdb/cockroach/pkg/storage/concurrency.(*workloadExecutor).execute.func1(0xc0001e4768, 0x22797469726f6874)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/concurrency/lock_table_test.go:779 +0x61 fp=0xc001b71f58 sp=0xc001b71f08 pc=0x1bcf611
github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc002d48120, 0xc0030e37d0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x72 fp=0xc001b71fd0 sp=0xc001b71f58 pc=0x1ba0202
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc001b71fd8 sp=0xc001b71fd0 pc=0xc11cf1
created by github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x74
```

The stacktrace is a little strange because of the locking in `(*lockTableImpl).findNextLockAfter`, but the root of the issue is that somehow `(*SpanSet).GetSpans(...)` panicked.

**Comment by sumeerbhola** [comment]
Could the panic in `(*SpanSet).GetSpans(...)` be due to an unclean shutdown due to some other panic, like the one fixed in #45124?
I am not able to reproduce this.

**Comment by sumeerbhola** [comment]
Closing since root cause is unknown, failure is not reproducible, and other bugs have been fixed in this code. Will reopen if occurs again.
